src/leap/bitmask/vpn/helpers/linux/bitmask-root

```diff
@@ -704,6 +727,28 @@ def firewall_start(args):
     ip4tables("--append", BITMASK_CHAIN, "-o",
               default_device, "--jump", "REJECT")
 
+    # On Qubes OS, add anti-leak rules for proxyVM qubes-firewall.service
+    # Must stay on 'top' of chain!
+    if QUBES_PROXY and QUBES_VER >= 3 and run("grep", \
+        "installed\ by\ " + SCRIPT, QUBES_FW_SCRIPT, exitcode=True) != 0:
+        with open(QUBES_FW_SCRIPT, mode="w") as qfile:
```

Comment:
This override the script, discarding previous its content. Is it intended?

Reply:
Yes. Do you have a better suggestion for managing this?

Reply:
Append? (`mode=a`)
You use `iptables --insert` anyway, so adding those commands at the end shouldn't be a problem. And you already avoid adding it twice, so that part should be ok too.
The only thing is to put shebang at the beginning, if the file didn't existed before.

Reply:
I thought about this but felt it assumed too much about the contents of a pre-existing script -- that there are no `exit` or other flow issues. That could compromise the anti-leak measures. I know of no decent way to mix arbitrary script sources into a larger, properly-ordered script (maybe for simple iptables rule sets, but not for sh). I don't even think sorted *.d script folders adequately address the need here.

Its one of the reasons I suggested moving that block to the "install helper files" code because that's an area where the user can be notified and prompted about files being overwritten in /rw/config.

Note that this issue is avoided in Qubes VPN doc and Qubes-vpn-support project by instructing the user to first create a new proxyVM to run the VPN solution.